src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
```

Comment:
Consider using an interface type for the 'storage' field (e.g., 'StorageDao') instead of a concrete implementation 'StorageDaoImpl'. This promotes the use of dependency inversion and makes the code more flexible and easier to test.